internal/types/infrastructure/requests.go

```diff
@@ -14,20 +16,41 @@ type InstancesRequest struct {
 	WebhookURL   string            `json:"webhook_url"`
 	Action       string            `json:"action"`
 	ProjectName  string            `json:"project_name"`
-	Provider     models.ProviderID `json:"provider"`
+	Provider     string            `json:"provider"`
 }
 
 // InstanceRequest represents a request to create or modify a compute instance
 type InstanceRequest struct {
-	Name              string            `json:"name"`                // Name of the instance
-	Provider          models.ProviderID `json:"provider"`            // Provider of the compute service
-	NumberOfInstances int               `json:"number_of_instances"` // Number of instances to create
-	Provision         bool              `json:"provision"`           // Whether to provision the instance
-	Region            string            `json:"region"`              // Region of the instance
-	Size              string            `json:"size"`                // Size of the instance
-	Image             string            `json:"image"`               // Image of the instance
-	Tags              []string          `json:"tags"`                // Tags of the instance
-	SSHKeyName        string            `json:"ssh_key_name"`        // SSH key name of the instance
+	Provider          string               `json:"provider"`            // Cloud provider (e.g., "digitalocean")
+	Region            string               `json:"region"`              // Region where instances will be created
+	Size              string               `json:"size"`                // Instance size/type
+	Image             string               `json:"image"`               // OS image to use
+	SSHKeyName        string               `json:"ssh_key_name"`        // Name of the SSH key to use
+	Tags              []string             `json:"tags"`                // Tags to apply to instances
+	NumberOfInstances int                  `json:"number_of_instances"` // Number of instances to create
+	Name              string               `json:"name"`                // Optional custom name for instances
+	Provision         bool                 `json:"provision"`           // Whether to run Ansible provisioning
+	Volumes           []types.VolumeConfig `json:"volumes"`             // Optional volumes to attach
+}
+
+// VolumeRequest represents the volume configuration in a request
+type VolumeRequest struct {
+	Name       string `json:"name"`        // Name of the volume
+	SizeGB     int    `json:"size_gb"`     // Size in gigabytes
+	Region     string `json:"region"`      // Region where to create the volume
```

Comment:
same, i don't think this is required because how can a volume be in a different region than the instance it is on.